File: Examples/test-suite/perl5/li_cstring_runme.pl
use strict;
use warnings;
use Test::More tests => 11;
BEGIN { use_ok('li_cstring') }
require_ok('li_cstring');

is(li_cstring::count("ab\0ab\0ab\0", 0), 3);
is(li_cstring::test1(), 'Hello World');
is(li_cstring::test2(), ' !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_');
is(li_cstring::test3('hello'), 'hello-suffix');
is(li_cstring::test4('hello'), 'hello-suffix');
is(li_cstring::test5(4), 'xxxx');
is(li_cstring::test6(10), 'xxxxx');
is(li_cstring::test7(), 'Hello world!');
is(li_cstring::test8(), ' !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_');

File: Examples/test-suite/matlab/li_cstring_runme.m
if (li_cstring.count('ab\0ab\0ab\0', 0) ~= 3)
    error    
end

if (~strcmp(li_cstring.test1(),'Hello World'))
    error
end

if (~strcmp(li_cstring.test2(),' !\'#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_'))
    error
end

if (~strcmp(li_cstring.test3('hello'),'hello-suffix'))
    error(test3('hello'))
end

if (~strcmp(li_cstring.test4('hello'),'hello-suffix'))
    error(li_cstring.test4('hello'))
end
    
if (~strcmp(li_cstring.test5(4),'xxxx'))
    error
end

if (~strcmp(li_cstring.test6(10),'xxxxx'))
    error
end
    
if (~strcmp(li_cstring.test7(),'Hello world!'))
    error
end

if (~strcmp(li_cstring.test8(),' !\'#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_'))
    error
end

